### PR TITLE
Fix links in docs

### PR DESCRIPTION
**Pull request by steakunderscore**
## Description

- Fixed a bunch of references to the repo, which were 404ing
- Fixed a couple of things that 301/302ed
- Fixed some in page references

## Motivation and Context

Well... they were broken ;)

## How Has This Been Tested?

Poked a couple of the links running `make serve` locally.

## Checklist:

- [ ] My change requires a change to the documentation or CHANGELOG.
- [ ] I have updated the documentation/CHANGELOG accordingly.
- [X] I have created a feature (non-master) branch for my PR.

### CODE REVIEW

File: docs/0_index.md [hunk 1]
```diff
@@ -12,12 +12,12 @@ to validate accounts by email, domain or group.
 
 **Note:** This repository was forked from [bitly/OAuth2_Proxy](https://github.com/bitly/oauth2_proxy) on 27/11/2018.
 Versions v3.0.0 and up are from this fork and will have diverged from any changes in the original fork.
-A list of changes can be seen in the [CHANGELOG](CHANGELOG.md).
+A list of changes can be seen in the [CHANGELOG]({{ site.gitweb }}/CHANGELOG.md).
```
Comment (syscll) [inline docs/0_index.md]: Is this a typo? :sweat_smile:

Comment (steakunderscore) [reply docs/0_index.md]: Is what a typo? gitweb is set in the [config](https://github.com/pusher/oauth2_proxy/pull/241/files#diff-b09bbff3e688a10b35f7de810d65c28eR21)

Comment (syscll) [reply docs/0_index.md]: Sorry, it's not correctly showing the markdown for me so I was unsure if this was a typo: 

image

Comment (steakunderscore) [reply docs/0_index.md]: It's using Jekyll syntax, for https://pusher.github.io/oauth2_proxy. There's already things that aren't supported by github markdown, like if you see the "table" at the top of the page.

**Pull request merged**